pkg/fs/fs_test.go

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fs
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestTempDir(t *testing.T) {
+	tempDir := os.TempDir()
+	tests := []struct {
+		name          string
+		dir           string
+		prefix        string
+		expectedError bool
+	}{
+		{
+			name:          "all is right",
+			dir:           tempDir,
+			prefix:        "tmp",
+			expectedError: false,
+		},
+		{
+			name:          "dir is empty",
+			dir:           "",
+			prefix:        "tmp",
+			expectedError: false,
+		},
+		{
+			name:          "dir doesn't exist",
+			dir:           "/abc",
+			prefix:        "tmp",
+			expectedError: true,
+		},
+		{
+			name:          "prefix is empty",
+			dir:           tempDir,
+			prefix:        "",
+			expectedError: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			name, err := TempDir(tt.dir, tt.prefix)
+			os.Remove(name)
+			if (nil != err) != tt.expectedError {
+				t.Errorf("expectedError: %v, got: %v", tt.expectedError, err)
+			}
+		})
+	}
+}
+
+func TestIsAbs(t *testing.T) {
```

Comment:
again we only seem to be testing the standard library behavior instead of our special behavior